packages/gatsby-plugin-mdx/src/compile-mdx.ts

```diff
@@ -1,27 +1,108 @@
+import type { NodePluginArgs } from "gatsby"
 import type { ProcessorOptions } from "@mdx-js/mdx"
-import type { IFileNode, IMdxNode } from "./types"
+import type { IFileNode, IMdxMetadata, IMdxNode } from "./types"
+
+import deepmerge from "deepmerge"
```

Comment:
Nit: Moving this import to the "gatsby" import and remove the blank lines between the imports